lib/modules/manager/azure-pipelines/extract.ts

```diff
@@ -13,7 +14,32 @@ const AzurePipelinesTaskRegex = regEx(/^(?<name>[^@]+)@(?<version>.*)$/);
 export function extractRepository(
   repository: Repository
 ): PackageDependency | null {
-  if (repository.type !== 'github') {
+  let repositoryUrl = null;
+
+  if (repository.type === 'github') {
+    repositoryUrl = `https://github.com/${repository.name}.git`;
+  } else if (repository.type === 'git') {
+    // "git" type indicates an AzureDevOps repository.
+    // The repository URL is only deducible if we are running on AzureDevOps (so can use the endpoint)
+    // and the name is of the form `Project/Repository`.
+    // The name could just be the repository name, in which case AzureDevOps defaults to the
+    // same project, which is not currently accessible here. It could be deduced later by exposing
+    // the repository URL to managers.
+    // https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/resources-repositories-repository?view=azure-pipelines#types
+    const { platform, endpoint } = GlobalConfig.get();
+    if (
+      platform === 'azure' &&
+      endpoint !== null &&
+      repository.name.includes('/')
```

Comment:
Not necessary, but maybe add a debug log here that by adding the project name, Renovate would be able to address this.